### PR TITLE
Logging fix

### PR DESCRIPTION
**Pull request by 2-am-zzz**
There is a new value in app/scripts/config.js called developmentMode that determines whether certain logging functions are enabled. Currently set to true--set to false for production.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -93,7 +94,7 @@ module.exports = class MetamaskController {
 
     function logger (err, request, response) {
       if (err) return console.error(err)
-      if (!request.isMetamaskInternal) {
+      if (developmentMode && !request.isMetamaskInternal) {
         console.log(`RPC (${originDomain}):`, request, '->', response)
         if (response.error) {
           console.error('Error in RPC response:\n', response.error)
```
Comment (flyswatter) [inline app/scripts/metamask-controller.js]: Logging errors actually doesn't need to be only in development, and I think is appropriate for production.

File: package.json [hunk 2]
```diff
@@ -35,6 +35,7 @@
     "debounce": "^1.0.0",
     "dnode": "^1.2.2",
     "end-of-stream": "^1.1.0",
+    "envify": "^3.4.1",
```
Comment (flyswatter) [inline package.json]: Unused dependency?

File: ui/app/store.js [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
 
 module.exports = configureStore
 
-const loggerMiddleware = createLogger()
+const middlewares = [thunkMiddleware]
 
-const createStoreWithMiddleware = applyMiddleware(
-  thunkMiddleware,
-  loggerMiddleware
+console.log(developmentMode)
```
Comment (flyswatter) [inline ui/app/store.js]: stray log

File: ui/app/store.js [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
+if (developmentMode) {
+  console.log('Development Mode ON.')
+  const createLogger = require('redux-logger')
+  const loggerMiddleware = createLogger()
+  middlewares.push(loggerMiddleware)
+} else {
+  console.log('Development Mode OFF.')
```
Comment (flyswatter) [inline ui/app/store.js]: Don't log this in production.

http://www.linfo.org/rule_of_silence.html

File: ui/app/store.js [hunk 5]
```diff
@@ -1,16 +1,25 @@
...
+}
+
+
+const createStoreWithMiddleware = applyMiddleware(...middlewares
```
Comment (flyswatter) [inline ui/app/store.js]: I feel like either this should all be on one line, or `...middlewares` should get its own line.

**Comment by Zanibas** [comment]
Thanks @flyswatter and @kumavis!

Now attempting to implement @kumavis suggestion of using https://github.com/evgenyrodionov/redux-logger#level-string--function--object

**Comment by flyswatter** [comment]
Yeah I think the easy solution is for the UI and background to have their own `METAMASK_DEBUG` vars for their own console logs.

When we adopt env vars in our gulp build, we can unite them.

**Comment by Zanibas** [comment]
@flyswatter does this all look good?

**Comment by flyswatter** [comment]
LGTM

**Comment by Zanibas** [comment]
For future reference: level will only change the actual nature of the messages--it is not a determination of whether the actual message gets logged (though you can gut the inside of the message!).

screenshot from 2016-08-11 18 37 48

**Pull request merged**